src/hdf5_back_gen.py

```diff
@@ -0,0 +1,496 @@
+#!/usr/bin/env python
+"""This module generates HDF5 backend code found in src/hdf5_back.cc"""
+import json
+import textwrap
+
+with open('../share/dbtypes.json') as f:
+    RAW_TABLE = json.load(f)
+
+V3_TABLE = set(tuple(row) for row in RAW_TABLE[897:])
+
+def list_dep(orig):
+    """Returns a list of dependent dbtypes.
+    
+    Keyword arguments: 
+    orig -- original dbtype.
+    
+    First element of returned list will always be orig itself.
+    """
+    pieces = orig.split("_")
+    if len(pieces) == 1:
+        return pieces
+    containers1 = ["VECTOR", "SET", "LIST", "VL_VECTOR", "VL_SET", "VL_LIST"]
+    containers2 = ["MAP", "PAIR", "VL_MAP", "VL_PAIR"]
+
+    if pieces[0] == "VL":
+        del pieces[0:2]
+    else:
+        del pieces[0]
+
+    for i in range(0, len(pieces)):
+        if pieces[i] == "VL":
+            pieces[i] = pieces[i] + "_" + pieces[i + 1]
+            del pieces[i + 1]
+        if i == len(pieces) - 1:
+            break
+    
+    for i in range(len(pieces) - 1, -1, -1):
+        if pieces[i] in containers1:
+            pieces[i] += "_" + pieces[i + 1]
+            del pieces[i + 1]
+        elif pieces[i] in containers2:
+            pieces[i]+= "_" + pieces[i + 1] + "_" + pieces[i + 2]
+            del pieces[i + 1:i + 3]
+    pieces = [orig] + pieces
+    return pieces
```

Comment:
The thing that confuses me is when we have the following:
```python
MAP_VL_STRING_X_READER =  """\
unsigned int itemsize = sizeof({t.sub[2].cpptype}) + CYCLUS_SHA1_SIZE;
jlen = col_sizes_[table][j] / itemsize;
{t.cpptype} x;
for (unsigned int k = 0; k < jlen; ++k) {{
    x[VLRead<{t.sub[1].cpptype}, {t.sub[1].dbtype}>(buf + offset + itemsize*k)] = *reinterpret_cast<{t.sub[2].cpptype}*>(buf + offset + itemsize*k + CYCLUS_SHA1_SIZE);
}}
{NO_CLOSE}\
"""
```
If we are generating sub types based upon C++ type, we know that {t.sub[1].dbtype} won't be accurate (since its dbtype in this case needs to be VL_STRING). VLRead would be taking in STRING, instead. Doesn't that pose a problem, since STRING is a completely different enum value?